### PR TITLE
implementing :+ and corrected hidden field inheritance in plus in jsonnet

### PR DESCRIPTION
**Pull request by SophiaSR**
Desugars :+ now (all that was needed to implement it) and fixed hidden field inheritance to use the correct hidden field based on spec.

PR checklist:

- [ ] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [ ] Tests included or PR comment includes a reproducible test plan
- [ ] Documentation is up-to-date
- [ ] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [ ] Change has no security implications (otherwise, ping security team)

If you're unsure about any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

File: libs/ojsonnet/Desugar_jsonnet.ml [hunk 1]
```diff
@@ -87,7 +87,7 @@ let freshvar =
     incr store;
     ("!tmp" ^ string_of_int !store, fk)
 
-let todo _env _v = failwith "TODO"
+(*let todo _env _v = failwith "TODO"*)
```
Comment (aryx) [inline libs/ojsonnet/Desugar_jsonnet.ml]: yeah! no more todos!

File: libs/ojsonnet/Unit_jsonnet_subst.ml [hunk 2]
```diff
@@ -3,6 +3,7 @@ open File.Operators
 module Y = Yojson.Basic
 
 let dir_pass = Fpath.v "tests/jsonnet/eval_pass"
```
Comment (aryx) [inline libs/ojsonnet/Unit_jsonnet_subst.ml]: At this point you can use a list of directories, like
let dir_pass = [
  "tests/jsonnet/eval_pass";
  "tests/jsonnet/tutorial/pass"
  ...
 ]  |> File.Path.of_strings

**Review by aryx — APPROVED** [review]
nice work!

File: libs/ojsonnet/Desugar_jsonnet.ml (hunk 1 above)
Comment (SophiaSR) [reply libs/ojsonnet/Desugar_jsonnet.ml]: :)

**Pull request merged**